ntoskrnl/mm/section.c

```diff
@@ -2003,7 +2005,13 @@ MmAccessFaultSectionView(PMMSUPPORT AddressSpace,
      * Unshare the old page.
      */
     DPRINT("Swapping page (Old %x New %x)\n", OldPage, NewPage);
-    MmDeleteVirtualMapping(Process, PAddress, NULL, NULL);
+    Unmapped = MmDeleteVirtualMapping(Process, PAddress, NULL, &UnmappedPage);
+    if (!Unmapped || (UnmappedPage != OldPage))
+    {
+        /* Uh , we had a page just before, but suddenly it changes. Someone corrupted us. */
+        KeBugCheck(MEMORY_MANAGEMENT);
```

Comment:
It would be of interest to use the Ex function and specify `(ULONG_PTR)__FILE__, __LINE__` as the last two parameters of the function, in order to help with debugging (since you don't use an assert here, we would loose some potential debugging info).